ogr/services/github.py

```diff
@@ -425,13 +426,32 @@ def get_issue_info(self, issue_id: int) -> Issue:
         issue = self.__get_issue(number=issue_id)
         return self._issue_from_github_object(issue)
 
-    def _get_all_issue_comments(self, issue_id: int) -> List[IssueComment]:
+    def get_issue_comments(
+        self, issue_id: int, filter_regex: str = None, reverse: bool = False
+    ) -> List[IssueComment]:
+        """
+        Get list of Issue comments.
+
+        :param issue_id: int
+        :param filter_regex: filter the comments' content with re.search
+        :param reverse: reverse order of comments
+        :return: [IssueComment]
+        """
         issue = self.__get_issue(number=issue_id)
-        return [
+        if not filter_regex:
+            filter_regex = r".*"
+
+        comments = [
             self._issuecomment_from_github_object(raw_comment)
-            for raw_comment in issue.get_issue_comments()
+            for raw_comment in issue.get_comments()
+            if re.search(filter_regex, raw_comment.body)
         ]
 
+        if reverse:
+            comments.reverse()
+
+        return comments
```

Comment:
Since the code is nearly the same for all implementation, what about:

- Implement skeleton implementation in the base class and use e.g. `_get_all_issue_comments` for getting all issue comments. (As we do for pull-request.) This can lead to spaghetti code.
- Add some filtering/sorting-helper method in the base class (or in some other place) and use them in all implementations.

Reply:
I planned to do that in #230, but ok

Reply:
Ok, so no problem.

Reply:
Thanks for fixing that here.

Reply:
So, revert it? And start working on #230?

Reply:
Do it as you wish. (In the current state, we only need to make the pre-commit happy.)

Reply:
Ok, I'll revert and push...
Probably stash what I have for #230...

Reply:
OK, choose the way you want.